Day 3 - Datetime strftime and strptime/Q2_Solution.py

```diff
@@ -0,0 +1,12 @@
+# 2. Convert the following date string to datetime object.
+#    --> "Jan 20, 2019 9:30 AM"
+
+from datetime import datetime
+
+
+
+inputDate = datetime(year=2019, day=20, month=1, hour=9, minute=30)
+print(datetime.strftime(inputDate, "%b %d, %Y %-H:%M %p"))
```

Comment:
There is an error here. `%-H`.

Reply:
Nope, it'll not throw any error, because by default `%H` will give output as `09`, the expected time output  has `9:30`, therefore to neglect `0`, so used `%-H`.

Hope this is clear.

Reply:
Okay. I should have explained better. My issue is not with `-`. my issue is with the use of `%H` part.
Why?
`%H` is a 24-hour clock as a zero-padded number. When we are using AM/PM in our date time strings, we expect our clock to be a 12 hour clock. For example, `14:30 PM` doesn't really make sense. To use a 12 hour clock, you should use `%I`.
Look at the below examples.
```
# Using %H

from datetime import datetime
inputDate = datetime(year=2019, day=20, month=1, hour=14, minute=30)
print(datetime.strftime(inputDate, "%b %d, %Y %-H:%M %p"))
```
This will return 
```
Jan 20, 2019 14:30 PM
```
To me, having a 24 hour clock with AM/PM details doesn't make sense. You got lucky that the hour was 9. Hence your solution looks okay. But its not going to be the case all the time.

```
# Using %I
from datetime import datetime
inputDate = datetime(year=2019, day=20, month=1, hour=14, minute=30)
print(datetime.strftime(inputDate, "%b %d, %Y %-I:%M %p"))
```
This snippet will return you,
```
Jan 20, 2019 2:30 PM
```
This makes sense to me. 

You might think this small change that I am asking you to consider is trivial. But when it comes to working with unlabelled datasets, using the right formatting techniques are highly essential. Also anyone else who reads it, might get confused when they see `%H` being used for 12 hour clock. 

Hope this explanation is clear.

Reply:
Moreover, you didn't have to do the first part. You are first creating a datetime object and converting it to a string. Then you don't use it in code. You create your own `InputString` variable. and then convert it to datetime object.

Reply:
Yeah, thanks for the good explanation. Will be removing the first part and also, 12 hour clock will be included.

Reply:
Actually, Mostapha has already provided the solution. Sorry about this time. but I will accept his solution for Day 3. Don't get discouraged and keep participating. You can close this PR for now. And please run a `rebase` on your repository. So you get all the changes, once I have updated.

Reply:
Sure, thanks for the motivation.